### PR TITLE
feat: add yml n json files for lighthouse CICD

### PR DESCRIPTION
**Pull request by isaackps**
## Summary

Add lighthouse to the project to improve the website.

## Changes

- added lighthouse.yml for github actions
- added lighthouse-budgets.json to set the score to criteria

## Issues

https://www.pivotaltracker.com/n/projects/2424361/stories/184411171

### CODE REVIEW

File: lighthouserc.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      "url": [
+        "http://localhost:3000/",
+        "http://localhost:3000/verify",
+        "http://localhost:3000/faq",
+        "http://localhost:3000/eta",
+        "http://localhost:3000/news",
+        "http://localhost:3000/learn",
+        "http://localhost:3000/event",
+        "http://localhost:3000/guidelines"
+      ],
```
Comment (simboonlong) [inline lighthouserc.json:27]: use glob pattern

File: lighthouserc.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+  "ci": {
+    "upload": {
+      "target": "temporary-public-storage"
+    },
```
Comment (simboonlong) [inline lighthouserc.json]: not sure if this is might be of [concern](https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/services-disclaimer.md#temporary-public-storage), or is there anyway to make use of the assertions without uploading reports?

File: lighthouserc.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "assert": {
+      "assertions": {
+        "categories:performance": ["warn", { "minScore": 0.6 }],
```
Comment (simboonlong) [inline lighthouserc.json:9]: i notice performance fluctuates, probably due to server resources allocated on each run. are the remaining 4 categories consistent?

File: lighthouserc.json (hunk 1 above)
Comment (isaackps) [reply lighthouserc.json:26]: what do u mean? got example?

File: lighthouserc.json (hunk 3 above)
Comment (isaackps) [reply lighthouserc.json:9]: yes, i notice that too, and yes probably due to the server resources allocated, even for dev.tt.io and tt.io they are different. 
So, for the PWA, our website is not PWA so i think it is irrelevant.
The other 3 are consistent, from what i have tested the other day.

File: lighthouserc.json (hunk 2 above)
Comment (isaackps) [reply lighthouserc.json]: Regarding this, u can have a look at this: https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/configuration.md#target

I used the quick and easy solution, that will delete the report in a few days.

So, the other option is using the 'lhci' option where we will upload the report into the lighthouse ci server, but need to setup a server and need to make access to anyone who wants the report.

You can take a look at this for the 'lhci' option.
https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/server.md

File: lighthouserc.json (hunk 1 above)
Comment (simboonlong) [reply lighthouserc.json:27]: wdyt to use .[js](https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/configuration.md#file-structure) as config instead and import the `routes` into `url `?

File: lighthouserc.json (hunk 3 above)
Comment (simboonlong) [reply lighthouserc.json:9]: ok sure, we can remove pwa assert or leave a comment that we not using that

Comment (isaackps) [reply lighthouserc.json:9]: ok sure

File: lighthouserc.json (hunk 2 above)
Comment (isaackps) [reply lighthouserc.json]: ok i just updated to make it save the reports as an artifact. 
Its a bit troublesome thou, need download it, unzip and open it in chrome to view the report.

Comment (simboonlong) [reply lighthouserc.json]: thats fine

File: .github/workflows/lighthouse.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        run: |
+          npm install -g @lhci/cli@0.10.x
+          lhci autorun
+      - name: upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: Lighthouse reports
+          path: ${{ github.workspace }}/lighthouseReport
+          if-no-files-found: ignore
+          retention-days: 5
```
Comment (simboonlong) [inline .github/workflows/lighthouse.yml]: remove `if-no-files-found` let it defaults to `warn`? retention 1 day

Comment (isaackps) [reply .github/workflows/lighthouse.yml]: ok

**Review by simboonlong — APPROVED** [review]

**Pull request merged**